lib/active_model/serializer/associations.rb

```diff
@@ -84,17 +84,22 @@ def associate(reflection)
       # @return [Enumerator<Association>]
       #
       def associations(include_directive = ActiveModelSerializers.default_include_directive)
-        return unless object
-
         Enumerator.new do |y|
-          self.class._reflections.values.each do |reflection|
-            next if reflection.excluded?(self)
-            key = reflection.options.fetch(:key, reflection.name)
-            next unless include_directive.key?(key)
-            y.yield reflection.build_association(self, instance_options)
+          self.class._reflections.keys.each do |association_key|
+            assoc = association(association_key, include_directive)
+            next if assoc.nil?
+            y.yield assoc
           end
         end
       end
+
+      def association(association_key, include_directive = ActiveModelSerializers.default_include_directive)
```

Comment:
not required in this PR, and only tested implicitly, but seems to make using associations a little easier